test/onnx/test_op_consistency.py

```diff
@@ -0,0 +1,462 @@
+
+    def forward(self, *args):
+        return self.operator(*args, **self.kwargs)
+
+
+class TestConsistency(common_utils.TestCase):
+    """Test consistency of exported ONNX models.
+
+    This is a parameterized test suite.
+    """
+
+    @classmethod
+    def create_test_base(cls, opset: int):
+        """Returns the base test method for the given opset."""
+
+        def _output_match_base(self, device: str, dtype: torch.dtype, op):
+            """Base test method for testing each opset."""
+            assert device == "cpu"
+
+            samples = op.sample_inputs(
+                device,
+                dtype,
+                requires_grad=False,
+            )
+
+            for (i, cpu_sample) in enumerate(samples):
+                # Provide the repr to subtest because tensors are not serializable in parallel test runs
+                with self.subTest(
+                    opset=opset,
+                    sample_num=i,
+                    input=repr(cpu_sample.input),
+                    args=repr(cpu_sample.args),
+                    kwargs=repr(cpu_sample.kwargs),
+                ):
+                    model = SingleOpModel(op, cpu_sample.kwargs)
+                    model.eval()
+
+                    # Run the test
+                    inputs = (cpu_sample.input, *cpu_sample.args)
+
+                    if dtype == torch.bfloat16:
+                        # Only export to ONNX without running with onnxruntime because
+                        # the CPU execution path for bfloat16 is not implemented in onnxruntime.
+                        model_buffer = io.BytesIO()
+                        torch.onnx.export(
+                            model, inputs, model_buffer, opset_version=opset
+                        )
+                        model_buffer.seek(0)
+                        onnx_model = onnx.load(model_buffer)
+                        onnx.checker.check_model(onnx_model, full_check=True)
+                        continue
+
+                    verification.verify(
+                        model,
+                        inputs,
+                        input_kwargs={},
+                        opset_version=opset,
+                        keep_initializers_as_inputs=True,
+                        ort_providers=ORT_PROVIDERS,
+                        check_shape=True,
+                        check_dtype=True,
+                        flatten=True,
+                    )
+
+        return _output_match_base
+
+    @classmethod
+    def parameterize_opsets(cls, opsets: Sequence[int]):
+        """Parameterizes the TestConsistency class with the given opsets."""
+        for opset in opsets:
+            # Generate a test method for each opset
+            test_name = f"test_output_match_opset_{opset}"
+            base_method = cls.create_test_base(opset)
+            # Important to rename the test method so that DecorateInfo can find it
+            base_method.__name__ = test_name
```

Comment:
`create_test_base` could have created such test name and free callers from having to hack it?

Reply:
Done